cmd/frontman/main.go

```diff
@@ -137,245 +165,315 @@ sudo sysctl -w net.ipv4.ping_group_range="0   2147483647"`
+		os.Exit(1)
+	}
+
+	os.Exit(0)
+}
+
+func handleFlagServiceInstall(fm *frontman.Frontman, serviceInstallUserPtr *string, serviceInstallPtr *bool, cfgPath string) {
+	// serviceInstallPtr is currently used on windows
+	// serviceInstallUserPtr is used on other systems
+	// if both of them are empty - just return
+	if (serviceInstallUserPtr == nil || *serviceInstallUserPtr == "") &&
+		(serviceInstallPtr == nil || !*serviceInstallPtr) {
+		return
+	}
+
+	systemManager := service.ChosenSystem()
+
+	username := ""
+	if serviceInstallUserPtr != nil {
+		username = *serviceInstallUserPtr
+	}
+
+	s, err := getServiceFromFlags(fm, cfgPath, username)
+	if err != nil {
+		fmt.Println(err.Error())
+		os.Exit(1)
+	}
+
+	if fm.HubURL == "" {
+		fmt.Printf("To install the service you first need to set 'hub_url' config param")
+		os.Exit(1)
+	}
+
+	if runtime.GOOS != "windows" {
+		userName := *serviceInstallUserPtr
+		u, err := user.Lookup(userName)
 		if err != nil {
-			log.Fatal(err)
+			fmt.Printf("Failed to find the user '%s'\n", userName)
+			os.Exit(1)
 		}
 
-		if service.Interactive() {
+		svcConfig.UserName = userName
+		// we need to chown log file with user who will run service
+		// because installer can be run under root so the log file will be also created under root
+		err = chownFile(fm.LogFile, u)
+		if err != nil {
+			fmt.Printf("Failed to chown log file for '%s' user\n", userName)
+		}
+	}
+	const maxAttempts = 3
+	for attempt := 1; attempt <= maxAttempts; attempt++ {
+		err = s.Install()
+		if err != nil && strings.Contains(err.Error(), "already exists") {
+			fmt.Printf("Frontman service(%s) already installed: %s\n", systemManager.String(), err.Error())
+
+			if attempt == maxAttempts {
+				fmt.Printf("Give up after %d attempts\n", maxAttempts)
+				os.Exit(1)
+			}
 
-			if *serviceUninstallPtr {
+			osSpecificNote := ""
+			if runtime.GOOS == "windows" {
+				osSpecificNote = " Windows Services Manager app should not be opened!"
+			}
+			if askForConfirmation("Do you want to overwrite it?" + osSpecificNote) {
 				err = s.Stop()
 				if err != nil {
+					// lets try to uninstall despite of this error
 					fmt.Println("Failed to stop the service: ", err.Error())
 				}
 
-				err = s.Uninstall()
-
+				err := s.Uninstall()
 				if err != nil {
-					fmt.Println("Failed to uninstall the service: ", err.Error())
+					fmt.Println("Failed to unistall the service: ", err.Error())
+					os.Exit(1)
 				}
-				return
 			}
 
-			if inputFilePtr != nil && *inputFilePtr != "" {
-				fmt.Println("Input file(-i) flag can't be used together with service install(-s) flag")
-				return
-			}
-			if outputFilePtr != nil && *outputFilePtr != "" {
-				fmt.Println("Output file(-o) flag can't be used together with service install(-s) flag")
-				return
-			}
-			if runtime.GOOS != "windows" {
+		} else if err != nil {
+			fmt.Printf("Frontman service(%s) installing error: %s\n", systemManager.String(), err.Error())
+			os.Exit(1)
+		} else {
```

Comment:
Do we need the ´else´?
I think all errors are handled inside the `if` branches.
So the normal behaviour is to `break` in the end.

Reply:
Otherwise, we will need to put `continue` in 2 error cases